Package.swift

```diff
@@ -249,7 +249,10 @@ let package = Package(
                 "Realm/RLMUserAPIKey.mm"
             ],
             publicHeadersPath: "include",
-            cxxSettings: cxxSettings
+            cxxSettings: cxxSettings,
+            linkerSettings: [
+                .linkedFramework("UIKit", .when(platforms: [.iOS, .macCatalyst, .tvOS, .watchOS]))
```

Comment:
@tgoyne I didn't add visionOS to this, given that we don't add support for visionOS until this PR is merged https://github.com/realm/realm-swift/pull/8309/. Is there a way to add visionOS conditional to a Xcode version